extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/NotEnoughTemporaryStorageFault.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.msq.indexing.error;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import java.util.Objects;
+
+@JsonTypeName(NotEnoughTemporaryStorageFault.CODE)
+public class NotEnoughTemporaryStorageFault extends BaseMSQFault
+{
+  static final String CODE = "NotEnoughTemporaryStorageFault";
+
+  private final long suggestedMinimumStorage;
+  private final long configuredTemporaryStorage;
+
+  @JsonCreator
+  public NotEnoughTemporaryStorageFault(
+      @JsonProperty("suggestedMinimumStorage") final long suggestedMinimumStorage,
+      @JsonProperty("configuredTemporaryStorage") final long configuredTemporaryStorage
+  )
+  {
+    super(
+        CODE,
+        "Not enough temporary storage space. Requires at least %,d bytes. (configured = %,d bytes)",
```

Comment:
We should also mention which runtime property to change here as well or disable durable storage using the context param.

Reply:
Updated